core/loadpoint_api.go

```diff
@@ -85,14 +94,14 @@ func (lp *LoadPoint) GetMinSoC() int {
 
 // SetMinSoC sets loadpoint charge minimum soc
 func (lp *LoadPoint) SetMinSoC(soc int) {
-	lp.Lock()
-	defer lp.Unlock()
-
 	lp.log.DEBUG.Println("set min soc:", soc)
 
 	// apply immediately
-	if lp.SoC.Min != soc {
+	if lp.GetMinSoC() != soc {
```

Comment:
Why to you check the GetMinSoc() in this SetMinSoC() setter? I can not see this behaviour in other setters which are protected now.

I think it might be a good idea to handle this very identical for all setters or document why we use a different behavior in that setter.